File: Algorithms/Bubble Sort/Bubblesort.java
public class BubbleSort {
	static void bubbleSort(int[] arr) {
        int n = arr.length;
        int temp = 0;
         for(int i=0; i < n; i++){
                 for(int j=1; j < (n-i); j++){
                          if(arr[j-1] > arr[j]){
                                 //swap elements
                                 temp = arr[j-1];
                                 arr[j-1] = arr[j];
                                 arr[j] = temp;
                         }

                 }
         }

    }
    void burbuja(int[] A) {
        int i, j, aux;
        for (i = 0; i < A.length - 1; i++) {
            for (j = 0; j < A.length - i - 1; j++) {
                if (A[j + 1] < A[j]) {
                    aux = A[j + 1];
                    A[j + 1] = A[j];
                    A[j] = aux;
                }
            }
        }
}
    public static void main(String[] args) {
                int arr[] ={3,60,35,2,45,320,5};

                System.out.println("Array Before Bubble Sort");
                for(int i=0; i < arr.length; i++){
                        System.out.print(arr[i] + " ");
                }
                System.out.println();

                bubbleSort(arr);//sorting array elements using bubble sort

                System.out.println("Array After Bubble Sort");
                for(int i=0; i < arr.length; i++){
                        System.out.print(arr[i] + " ");
                }

        }
}

File: Algorithms/Bubble Sort/bubble_sort.cpp
#include <iostream>
using namespace std;

void swap(int *f, int *l);

void bubbleSort(int numbers[], int n);

void printsortedarray(int numbers[], int n);

int main()
{
    int n ;
    cout<<"Enter no of elements: \n";
    cin>>n;

    int numbers[n];
    cout<<"Enter the elements: \n";

    for (int i = 0; i < n; ++i) {
        cin >> numbers[i];
    }
    bubbleSort(numbers, n);
    cout<<"Sorted array: \n";
    printsortedarray(numbers, n);
    return 0;
}

void swap(int *f, int *l)
{
    int temp = *f;
    *f = *l;
    *l = temp;
}

void bubbleSort(int numbers[], int n)
{
    int i, j;
    for (i = 0; i < n-1; i++)

    for (j = 0; j < n-i-1; j++)
        if (numbers[j] > numbers[j+1])
            swap(&numbers[j], &numbers[j+1]);
}

void printsortedarray(int numbers[], int n)
{
    int i;
    for (i = 0; i < n; i++)
        cout << numbers[i] << " ";
    cout << endl;
}
